### PR TITLE
add test for getPets function

### PR DESCRIPTION
**Pull request by Alexandru-Bulai**
# PR 15-02-24
## Overview
This PR introduces unit tests for the getPets() function within our application. The getPets() function is responsible for fetching pet data from a JSON file and dynamically populating the gallery container on the gallery.html page with pet elements## Key Changes :
 1) galleryDom.test.js :
    - Implementation of Mock Fetch: To isolate the tests from network dependencies, we've mocked the global fetch function using Jest. This approach allows us to simulate different responses from the server and ensure our function reacts appropriately.
    - Before Each Test Cleanup: A beforeEach hook is used to clear any mocks, ensuring that each test runs in a clean state without interference from the outcomes of previous tests.
    - Test for Handling Empty Data Response: Another test ensures that getPets()  handles scenarios where the fetch API returns an empty array of pets.

### CODE REVIEW

File: test/galleryDom.test.js [hunk 1]
```diff
@@ -48,3 +48,63 @@ describe('Gallery Counter', () => {
...
+      })
+    )
+
+    const dom = new JSDOM('<!DOCTYPE html><html><body><div id="img-container"></div></body></html>')
+    global.document = dom.window.document
+
+    await getPets('#img-container')
+
+    const galleryContainer = document.querySelector('#img-container')
+    expect(galleryContainer.children.length).toBe(0)
```
Comment (vladiiancu) [inline test/galleryDom.test.js:110]: Very nice

File: test/galleryDom.test.js [hunk 2]
```diff
@@ -48,3 +48,63 @@ describe('Gallery Counter', () => {
...
+    const firstItem = galleryItems[0]
+    expect(firstItem.dataset.type).toBe('dog')
+    expect(firstItem.querySelector('img').src).toContain('dog.jpg')
+    expect(firstItem.querySelector('.text-15.font-bold.text-blue-800').textContent).toContain('Gender: male')
+  })
+
+  it('should handle empty data response gracefully', async () => {
+    global.fetch.mockImplementationOnce(() =>
+      Promise.resolve({
+        json: () => Promise.resolve({ payload: { pets: [] } })
```
Comment (vladiiancu) [inline test/galleryDom.test.js:100]: Interesting. I think I like it

File: test/galleryDom.test.js [hunk 3]
```diff
@@ -48,3 +48,63 @@ describe('Gallery Counter', () => {
...
+    const dom = new JSDOM('<!DOCTYPE html><html><body><div id="img-container"></div></body></html>')
+    global.document = dom.window.document
+
+    await getPets('#img-container')
+
+    const galleryContainer = document.querySelector('#img-container')
+    const galleryItems = galleryContainer.querySelectorAll('.gallery-items')
+
+    // Check if the gallery items are correctly added to the galleryContainer
+    expect(galleryItems.length).toBeGreaterThan(0)
```
Comment (vladiiancu) [inline test/galleryDom.test.js]: The length depends on the json file right? 
If there are 5 items there, the length should be 5.

If the mocked json contains 1 element and the gallery function shows 10 times the same element, the test will pass but the behaviour is broken

**Review by vladiiancu — CHANGES_REQUESTED** [review]

File: test/galleryDom.test.js [hunk 4]
```diff
@@ -48,3 +48,65 @@ describe('Gallery Counter', () => {
...
+)
+
+describe('json file fetching data on gallery.html', () => {
+  beforeEach(() => {
+    // Reset fetch mock before each test
+    fetch.mockClear()
+  })
+
+  it('Populate galleryContainer with pet elements from JSON', async () => {
+    const dom = new JSDOM('<!DOCTYPE html><html><body><div id="img-container"></div></body></html>')
```
Comment (vladiiancu) [inline test/galleryDom.test.js:77]: Very nice. Minimal required. Good.

**Review by vladiiancu — APPROVED** [review]
